### PR TITLE
More tests for json and metrics

### PR DESCRIPTION
**Pull request by splitfeed**

### CODE REVIEW

File: json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java [hunk 1]
```diff
@@ -32,10 +42,69 @@ public void setUp() throws Exception {
...
+        Function<String, ImmutableEntity> deserializer = deserializerFactory.createDeserializer(ImmutableEntity.class);
+
+        ImmutableEntity result = deserializer.apply(null);
+
+        assertThat(result).isNull();
+    }
+
+
+    @Test
+    public void shouldshouldSerializeAndDeserializeImmutableObjectsByTypeReference() {
```
Comment (tsfcode) [inline json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java]: A superfluous should has snuck it's way into the functionname.

File: json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java [hunk 2]
```diff
@@ -32,10 +42,69 @@ public void setUp() throws Exception {
...
+
+        Function<byte[], ImmutableEntity> deserializer = deserializerFactory.createByteDeserializer(immutableEntityTypeReference);
+        ImmutableEntity result = deserializer.apply(json);
+
+        assertThat(result).isEqualTo(immutableEntity);
+    }
+
+
+	@Test
+	public void shouldshouldSerializeAndDeserializeImmutableObjectsByTypeReferenceToAndFromByteArrays() {
```
Comment (tsfcode) [inline json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java]: Double should here aswell.

File: json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java (hunk 1 above)
Comment (splitfeed) [reply json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java]: Fixed!

File: json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java (hunk 2 above)
Comment (splitfeed) [reply json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java]: Fixed!

File: json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java [hunk 3]
```diff
@@ -32,10 +42,69 @@ public void setUp() throws Exception {
...
+        ImmutableEntity result = deserializer.apply(json);
+
+        assertThat(result).isEqualTo(immutableEntity);
+    }
+
+
+	@Test
+	public void shouldSerializeAndDeserializeImmutableObjectsByTypeReferenceToAndFromByteArrays() {
+		Function<ImmutableEntity, String> serializer = serializerFactory.createStringSerializer(immutableEntityTypeReference);
+		String json = serializer.apply(immutableEntity);
```
[the remark this answers is not in the document]
Comment (splitfeed) [reply json/src/test/java/se/fortnox/reactivewizard/json/JsonDeserializerFactoryTest.java]: You're right, some tabs snuck in. Fixing!

**Pull request merged**